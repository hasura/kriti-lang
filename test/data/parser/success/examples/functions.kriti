# function calls
[ {{ escapeUri("foo") }},
  {{ escapeUri("foo{{$bar}}") }},
  {{ escapeUri($path[0]) }},
  {{ escapeUri("?foo=bar/baz") }},
  "http://www.google.com{{escapeUri($params)}}"
]