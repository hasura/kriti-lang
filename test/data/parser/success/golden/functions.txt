Array
    ( Span
        { start = AlexSourcePos
            { line = 1
            , col = 1
            }
        , end = AlexSourcePos
            { line = 6
            , col = 2
            }
        }
    )
    [ Function
        ( Span
            { start = AlexSourcePos
                { line = 1
                , col = 6
                }
            , end = AlexSourcePos
                { line = 1
                , col = 15
                }
            }
        ) "escapeUri"
        ( StringTem
            ( Span
                { start = AlexSourcePos
                    { line = 1
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 1
                    , col = 21
                    }
                }
            )
            [ String
                ( Span
                    { start = AlexSourcePos
                        { line = 1
                        , col = 17
                        }
                    , end = AlexSourcePos
                        { line = 1
                        , col = 20
                        }
                    }
                ) "foo"
            ]
        )
    , Function
        ( Span
            { start = AlexSourcePos
                { line = 2
                , col = 6
                }
            , end = AlexSourcePos
                { line = 2
                , col = 15
                }
            }
        ) "escapeUri"
        ( StringTem
            ( Span
                { start = AlexSourcePos
                    { line = 2
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 2
                    , col = 29
                    }
                }
            )
            [ String
                ( Span
                    { start = AlexSourcePos
                        { line = 2
                        , col = 17
                        }
                    , end = AlexSourcePos
                        { line = 2
                        , col = 20
                        }
                    }
                ) "foo"
            , Path
                ( Span
                    { start = AlexSourcePos
                        { line = 2
                        , col = 22
                        }
                    , end = AlexSourcePos
                        { line = 2
                        , col = 26
                        }
                    }
                )
                [ Obj
                    ( Span
                        { start = AlexSourcePos
                            { line = 2
                            , col = 22
                            }
                        , end = AlexSourcePos
                            { line = 2
                            , col = 26
                            }
                        }
                    ) NotOptional "$bar" Head
                ]
            ]
        )
    , Function
        ( Span
            { start = AlexSourcePos
                { line = 3
                , col = 6
                }
            , end = AlexSourcePos
                { line = 3
                , col = 15
                }
            }
        ) "escapeUri"
        ( Path
            ( Span
                { start = AlexSourcePos
                    { line = 3
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 3
                    , col = 24
                    }
                }
            )
            [ Obj
                ( Span
                    { start = AlexSourcePos
                        { line = 3
                        , col = 16
                        }
                    , end = AlexSourcePos
                        { line = 3
                        , col = 21
                        }
                    }
                ) NotOptional "$path" Head
            , Arr
                ( Span
                    { start = AlexSourcePos
                        { line = 3
                        , col = 21
                        }
                    , end = AlexSourcePos
                        { line = 3
                        , col = 24
                        }
                    }
                ) NotOptional 0
            ]
        )
    , Function
        ( Span
            { start = AlexSourcePos
                { line = 4
                , col = 6
                }
            , end = AlexSourcePos
                { line = 4
                , col = 15
                }
            }
        ) "escapeUri"
        ( StringTem
            ( Span
                { start = AlexSourcePos
                    { line = 4
                    , col = 16
                    }
                , end = AlexSourcePos
                    { line = 4
                    , col = 30
                    }
                }
            )
            [ String
                ( Span
                    { start = AlexSourcePos
                        { line = 4
                        , col = 17
                        }
                    , end = AlexSourcePos
                        { line = 4
                        , col = 29
                        }
                    }
                ) "?foo=bar/baz"
            ]
        )
    , StringTem
        ( Span
            { start = AlexSourcePos
                { line = 5
                , col = 3
                }
            , end = AlexSourcePos
                { line = 5
                , col = 47
                }
            }
        )
        [ String
            ( Span
                { start = AlexSourcePos
                    { line = 5
                    , col = 4
                    }
                , end = AlexSourcePos
                    { line = 5
                    , col = 25
                    }
                }
            ) "http://www.google.com"
        , Function
            ( Span
                { start = AlexSourcePos
                    { line = 5
                    , col = 27
                    }
                , end = AlexSourcePos
                    { line = 5
                    , col = 36
                    }
                }
            ) "escapeUri"
            ( Path
                ( Span
                    { start = AlexSourcePos
                        { line = 5
                        , col = 37
                        }
                    , end = AlexSourcePos
                        { line = 5
                        , col = 44
                        }
                    }
                )
                [ Obj
                    ( Span
                        { start = AlexSourcePos
                            { line = 5
                            , col = 37
                            }
                        , end = AlexSourcePos
                            { line = 5
                            , col = 44
                            }
                        }
                    ) NotOptional "$params" Head
                ]
            )
        ]
    ]